pkg/config/config.go

```diff
@@ -486,14 +490,29 @@ func AddAgentVersionToDomain(domain string, app string) (string, error) {
 	return u.String(), nil
 }
 
+// getMainEndpoint implements the logic to extract the main DD URL from a config
+func getMainEndpoint(config Config) (ddURL string) {
```

Comment:
Could it be possible to export it publicly and add an `agent prefix` as a parameter so it can be used from the other agents ?

Reply:
that'd make sense yes 👍  I'll have a look

Reply:
This ended up being more complex than what you described: see `GetMainEndpoint` in https://github.com/DataDog/datadog-agent/pull/2412/commits/9fb323939da96f5de67d32b044a83cfc9d83335a 

Even with this change, the `DD_PROCESS_AGENT_URL` env var would still need to be handled specifically in process_agent because it doesn't follow the regular env var naming scheme.

Let me know if you think this commit is useful (I'll remove it if it's not), or if you have any other comments!

Reply:
Thanks ! Since the logic for handling env variables and file variables is separated in the process-agent I think it would still be useful to use this helper for the file config 👍

Reply:
@sfluor sounds good. Please note that `GetMainEndpoint` takes into account the value of the `DD_SITE` env var (and the `DD_PROCESS_CONFIG_PROCESS_DD_URL` as well, even if this one's not really relevant)